react/SearchAction.tsx

```diff
@@ -19,21 +19,26 @@ declare var window: {
 }
 
 const SearchAction = () => {
+  const runtime = useRuntime()
   const protocol = 'https'
   const hostname = canUseDOM ? window.location.hostname : global.__hostname__
-  const rootPath = canUseDOM ? window.__RUNTIME__.rootPath : global.__RUNTIME__.rootPath
+  const rootPath = canUseDOM
+    ? window.__RUNTIME__.rootPath
+    : global.__RUNTIME__.rootPath
 
   const baseUrl = `${protocol}://${hostname}${rootPath || ''}/`
+  const { searchTerm } = runtime.getSettings('vtex.store')
```

Comment:
I think it's better to change to receiving this as props from vtex.store, because in the future it will be hard to track where this setting is being used and someone might remove it accidentally.